sirepo/template/radia.py

```diff
@@ -394,13 +420,18 @@ def _get_sdds():
     return _cfg.sdds
 
 
-def _read_data(sim_id, view_type, field_type):
+def _read_path(sim_id, h5path):
     try:
         with h5py.File(_geom_file(sim_id), 'r') as hf:
-            g = template_common.h5_to_dict(hf, path=_geom_h5_path(view_type, field_type))
-        return g
+            return template_common.h5_to_dict(hf, path=h5path)
     except IOError:
-        return {}
+        return PKDict()
+
+
+def _read_data(sim_id, view_type, field_type):
+    res = _read_path(sim_id, _geom_h5_path(view_type, field_type))
+    res.solution = _read_solution(sim_id)
```

Comment:
If read path fails, you shouldn't try to read the solution, and there shouldn't be a solution attribute. This is where errors need to be documented by if statements.